cmd/kubeadm/app/util/pkiutil/pki_helpers.go

```diff
@@ -383,6 +381,12 @@ func GetAPIServerAltNames(cfg *kubeadmapi.InitConfiguration) (*certutil.AltNames
 			advertiseAddress,
 		},
 	}
+	// cfg.NodeRegistration.Name may use ip!
```

Comment:
thank you for the PR,
i don't think i agree with this change and the similar one in getAltNames() . the comment that "node names can be IPs" is wrong.

the Node name is supposed to be a string "that can be used as a DNS subdomain name as defined in RFC 1123"
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

https://github.com/kubernetes/kubernetes/blob/47a9952337dada2cd57f3b2b3aeccac500277318/cmd/kubeadm/app/apis/kubeadm/validation/validation.go#L107

however, this function is not perfectly following RFC 1123 and it allows the strings to start with a number, which is wrong.

the node name *should be* a hostname, not an IP address and we should not parse it as an IP address.
if you want to add the same IP address in altNames you should do so explicitly using the extraSANs kubeadm feature.


/hold

Reply:
Thanks for pointing out the error, leave it as it is.